### DISCUSSION TITLE
How TextField support copy and paste

### DISCUSSION THREAD

**Discussion by Charliego3** [progrium/darwinkit#204] [Q&A] [answered]
How does the TextField support copy and paste events?

**Comment by progrium** [comment, marked as answer]
They will have copy/paste events if you have a menu that include those operations. See our menu example:
https://github.com/progrium/macdriver/blob/main/macos/_examples/menu/main.go

If you don't want a menu, you have to manually catch keypresses and do copy/paste operations:
```golang
f := appkit.NewTextField()
f.CurrentEditor().Copy(nil)
f.CurrentEditor().Paste(nil)
```
See our subclass example for one way to capture keypresses.
